src/transport/raw/TCP.cpp

```diff
@@ -47,13 +47,14 @@ constexpr int kListenBacklogSize = 2;
 /**
  *  Determine if the given buffer contains a complete message
  */
-bool ContainsCompleteMessage(const System::PacketBufferHandle & buffer, uint8_t ** start, uint16_t * size)
+bool ContainsCompleteMessage(const System::PacketBufferHandle & buffer, uint8_t ** start, uint16_t * size, uint16_t * payloadSize)
 {
     bool completeMessage = false;
 
     if (buffer->DataLength() >= kPacketSizeBytes)
     {
         *size           = LittleEndian::Get16(buffer->Start());
+        *payloadSize    = static_cast<uint16_t>(*size + kPacketSizeBytes);
         *start          = buffer->Start() + kPacketSizeBytes;
         completeMessage = (buffer->DataLength() >= *size + kPacketSizeBytes);
```

Comment:
Might be worth rewriting this as `(buffer->DataLength() - kPacketSizeBytes >= *size)` to avoid thinking about overflow there.